sql: panic on ALTER TABLE RENAME

Similar to #7863.

`> ALTER TABLE t RENAME TO t.*`

Panics the server:

```
panic: interface conversion: parser.IndirectionElem is parser.StarIndirection, not parser.NameIndirection

goroutine 229 [running]:
panic(0x18b13c0, 0xc82005d8c0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/cockroachdb/cockroach/sql/parser.(*QualifiedName).Table(0xc82005d880, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/parser/expr.go:753 +0x1d7
github.com/cockroachdb/cockroach/sql.(*planner).RenameTable(0xc8200cb090, 0xc82070c300, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/rename.go:112 +0xd9
github.com/cockroachdb/cockroach/sql.(*planner).newPlan(0xc8200cb090, 0x7f15ce5c6a48, 0xc82070c300, 0x0, 0x0, 0x0, 0x1b087c01, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/plan.go:270 +0x3a4
github.com/cockroachdb/cockroach/sql.(*planner).makePlan(0xc8200cb090, 0x7f15ce5c6a48, 0xc82070c300, 0x83fb01, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/plan.go:212 +0x83
github.com/cockroachdb/cockroach/sql.(*Executor).execStmt(0xc8200a6000, 0x7f15ce5c6a48, 0xc82070c300, 0xc8200cb090, 0xc82070c301, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:1045 +0x93
github.com/cockroachdb/cockroach/sql.(*Executor).execStmtInOpenTxn(0xc8200a6000, 0x7f15ce5c6a48, 0xc82070c300, 0xc8200cb090, 0x101, 0xc8200cb028, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:943 +0x439
github.com/cockroachdb/cockroach/sql.(*Executor).execStmtsInCurrentTxn(0xc8200a6000, 0xc82075c770, 0x1, 0x1, 0xc8200cb090, 0xc8200cb028, 0x1fe0101, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:691 +0x5da
github.com/cockroachdb/cockroach/sql.runTxnAttempt(0xc8200a6000, 0xc8200cb090, 0x1, 0xc8200cb028, 0xc82070c380, 0xc82075c770, 0x1, 0x1, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:619 +0x18d
github.com/cockroachdb/cockroach/sql.(*Executor).execRequest.func2(0xc8201c49a0, 0xc82070c380, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:511 +0x470
github.com/cockroachdb/cockroach/internal/client.(*Txn).Exec(0xc8201c49a0, 0xc820000101, 0x1461fe7771379393, 0x0, 0xc82000aba0, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/internal/client/txn.go:548 +0x2eb
github.com/cockroachdb/cockroach/sql.(*Executor).execRequest(0xc8200a6000, 0x7f15ce5c5ae0, 0xc820466f40, 0xc8200cb000, 0xc82048002d, 0x1b, 0x0, 0x0, 0x0, 0xc820480000)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:516 +0xba7
github.com/cockroachdb/cockroach/sql.(*Executor).ExecuteStatements(0xc8200a6000, 0x7f15ce5c5ae0, 0xc820466f40, 0xc8200cb000, 0xc82048002d, 0x1b, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/executor.go:382 +0xf6
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).executeStatements(0xc8207f0c00, 0x7f15ce5c5ae0, 0xc820466f40, 0xc82048002d, 0x1b, 0x0, 0x0, 0x0, 0x0, 0x7f15ce5c6501, ...)
	/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:640 +0x98
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).handleSimpleQuery(0xc8207f0c00, 0x7f15ce5c5ae0, 0xc820466f40, 0xc8207f0c28, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:320 +0xe8
github.com/cockroachdb/cockroach/sql/pgwire.(*v3Conn).serve(0xc8207f0c00, 0xc820469460, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/pgwire/v3.go:275 +0x100c
github.com/cockroachdb/cockroach/sql/pgwire.(*Server).ServeConn(0xc8201a6a50, 0x7f15ce5c6440, 0xc8202e8f00, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/sql/pgwire/server.go:229 +0x98f
github.com/cockroachdb/cockroach/server.(*Server).Start.func8.1(0x7f15ca53c2f0, 0xc820172000)
	/go/src/github.com/cockroachdb/cockroach/server/server.go:370 +0x42
github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith.func1(0xc8201aa238, 0x7f15ca53c2f0, 0xc820172000, 0xc8204d8030)
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:131 +0x62
created by github.com/cockroachdb/cockroach/util/netutil.(*Server).ServeWith
	/go/src/github.com/cockroachdb/cockroach/util/netutil/net.go:133 +0x333
```

Found with the random syntax generator.

--- a/sql/parser/expr.go
+++ b/sql/parser/expr.go
@@ -622,6 +622,12 @@ func (node *QualifiedName) NormalizeTableName(database string) error {
 	if len(node.Indirect) > 1 {
 		return fmt.Errorf("invalid table name: %s", node)
 	}
+	switch node.Indirect[0].(type) {
+	case NameIndirection:
+		// Nothing to do.
+	default:
+		return fmt.Errorf("invalid table name: %s", node)
+	}
 	node.normalized = tableName
 	return nil
 }

--- a/sql/parser/expr_test.go
+++ b/sql/parser/expr_test.go
@@ -76,6 +76,8 @@ func TestNormalizeTableName(t *testing.T) {
 		{`test.foo.bar`, ``, ``, `invalid table name: test.foo.bar`},
 		{`test.foo[bar]`, ``, ``, `invalid table name: test.foo\[bar\]`},
 		{`test.foo.bar[blah]`, ``, ``, `invalid table name: test.foo.bar\[blah\]`},
+		{`test.*`, ``, ``, `invalid table name: test.*`},
+		{`test[blah]`, ``, ``, `invalid table name: test\[blah\]`},
 	}
 
 	for _, tc := range testCases {

--- a/sql/testdata/alter_table
+++ b/sql/testdata/alter_table
@@ -10,6 +10,12 @@ EXPLAIN (DEBUG) SELECT * FROM t
 RowIdx Key          Value Disposition
 0      /t/primary/1 NULL  ROW
 
+statement error invalid table name
+ALTER TABLE t RENAME TO t.*
+
+statement error invalid table name
+ALTER TABLE t RENAME TO t[TRUE]
+
 statement ok
 ALTER TABLE t ADD b INT
 

--- a/sql/testdata/select
+++ b/sql/testdata/select
@@ -31,6 +31,18 @@ SELECT * FROM abc
 a b c
 1 2 3
 
+# synonym for SELECT * FROM abc
+query III
+TABLE abc
+----
+1 2 3
+
+query error invalid table name
+TABLE abc[TRUE]
+
+query error invalid table name
+TABLE abc.*
+
 query III colnames
 SELECT * FROM abc WHERE NULL
 ----

